### ISSUE TITLE
Error -32602 Sum of amounts of all outgoing transfers cannot be 0

### ISSUE THREAD

**Issue by zool2003** [bcndev/bytecoin#36]
Hi,

I have been connecting to the RPC and sending commands just fine with the exception of transfers.

I have modified the example to give you an idea of what it is that I am using to try and initiate a transfer.

curl -X POST http://<ip>:<port>/json_rpc -H 'Content-Type: application/json-rpc' -d '{"jsonrpc": "2.0", "id": "<id>", "method": "<method>", "params": {"change_address":"","spend_address":"","spend_address":"","amount":"1"}}'

Response 
{
  "error": {
    "code": -32602,
    "message": "Sum of amounts of all outgoing transfers cannot be 0"
  },
  "id": "0",
  "jsonrpc": "2.0"

I am using this through a Httpwebrequest .in VB

Sorry if I have missed something obvious but I have been stuck for 2 days trying to figure it out.

**Comment by virgeny** [comment]
It`s new walletd ? 
there are a new methods and little bit other parameters and methods
check on wiki 
https://github.com/bcndev/bytecoin/wiki/Bytecoin-Wallet-Daemon-JSON-RPC-API#input-params-9
it`s not well described, but in addition to that parameters you need to add array "transfers" with values amount and address  to transaction array
so
carefuly read all the **Create transaction** paragraph

**Comment by zool2003** [comment]
Yeah it’s the new wallets file and I have been working through the new wiki and have most other things working.

I have to say the new walletd with the built in node is much better.

But the documentation on the transfers could do with updating because it does suggest a transfers array with the addresses and amounts.

I will have another try tonight.

**Comment by FndNur1Labs** [comment]
This make single nodes run with wallet?

**Comment by zool2003** [comment]
Yes single node running from walletd. 

I can get a list of transfers and the balance ok just cannot transfer.

I think there needs to be an example of how this is done?

**Comment by FndNur1Labs** [comment]
Is this can make better on chain. What I know we need split supply chain by
account, data n log. Where nodes have 3 function self....as we know some
less bw can't full nodes. What we need change is task on chain split to
some layer

**Comment by zool2003** [comment]
@virgeny 

Tried to add "transfers":{} and I just get a parse error? to what I have and getting a parse error.

**Comment by zool2003** [comment]
Still haven't managed to work out what the structure needs to be to send a transfer. If anyone knows or can point me in the direction of where I can find it then please let me know.

**Comment by bcndev** [comment]
@zool2003 Initiation of a transfer seems to be nonsense wrt our model of money management. By transfer we mean the result of a transaction sending money from the sender to the receivers. So, you form transfers by creating transactions. List of transfers is available [here](https://github.com/bcndev/bytecoin/wiki/Bytecoin-Wallet-Daemon-JSON-RPC-API#8-get-transfers).

**Comment by bcndev** [comment]
@zool2003 Hope [these jsons](https://pastebin.com/G7brPeNY) will help you figure out how to create and send transactions.

**Comment by zool2003** [comment]
Those JSON commands were a massive help thank you. 

I was able to construct a transfer and I have a broadcast response but when I check the wallet balance nothing is sent. I am currently trying to use a newer version of the wallet to see
